pkg/infra/filestorage/api.go

```diff
@@ -0,0 +1,101 @@
+package filestorage
+
+import (
+	"context"
+	"errors"
+	"strings"
+	"time"
+)
+
+type StorageName string
+
+const (
+	StorageNamePublic StorageName = "public"
+)
+
+var (
+	ErrRelativePath          = errors.New("path cant be relative")
+	ErrNonCanonicalPath      = errors.New("path must be canonical")
+	ErrPathTooLong           = errors.New("path is too long")
+	ErrPathInvalid           = errors.New("path is invalid")
+	ErrPathEndsWithDelimiter = errors.New("path can not end with delimiter")
+	Delimiter                = "/"
+)
+
+func Join(parts ...string) string {
+	return Delimiter + strings.Join(parts, Delimiter)
+}
+
+func belongsToStorage(path string, storageName StorageName) bool {
```

Comment:
checking prefix seems just as easy... and perhaps more clear.

But does the path always start with a '/'?

Reply:
fixed delimiter check + added tests - I _think_ it's good to use absolute paths everywhere consistently